store.go

```diff
@@ -2351,9 +2519,14 @@ func (s *store) DeleteImage(id string, commit bool) (layers []string, err error)
 		}
 		if commit {
 			for _, layer := range layersToRemove {
-				if err = rlstore.Delete(layer); err != nil {
+				if err = rlstore.Delete(layer); err != nil && err != ErrLayerUnknown {
 					return err
 				}
+				if crlstore != nil {
+					if err = crlstore.Delete(layer); err != nil && err != ErrLayerUnknown {
+						return err
+					}
+				}
 			}
```

Comment:
This ends up completely ignoring `ErrLayerUnknown` errors…

I dunno, that might be the right thing to do, actually.